collector/lib/rust/host/src/info.rs

```diff
@@ -0,0 +1,80 @@
+use crate::KernelVersion;
+
+use std::fs::File;
+use std::io::{prelude::*, BufReader};
+use std::path::PathBuf;
+
+// The values are taken from efi_secureboot_mode in include/linux/efi.h
+#[derive(Default)]
+pub enum SecureBootStatus {
+    ENABLED = 3,
+    DISABLED = 2,
+    // Secure Boot seems to be disabled, but the boot loaded
+    // does not provide enough information about it,
+    // so it could be enabled without the kernel being aware.
+    #[default]
+    NOT_DETERMINED = 1,
+    UNSET = 0, // No detection is performed yet
+}
+
+pub struct HostInfo;
+
+impl HostInfo {
+    pub fn is_cos(&self) -> bool {
+        self.os_id().eq("cos") && !self.build_id().is_empty()
+    }
+
+    pub fn is_coreos(&self) -> bool {
+        self.os_id().eq("coreos")
+    }
+
+    pub fn is_docker_desktop(&self) -> bool {
+        self.os_id().eq("Docker Desktop")
+    }
+
+    pub fn is_ubuntu(&self) -> bool {
+        self.os_id().eq("ubuntu")
+    }
+
+    pub fn is_garden(&self) -> bool {
+        self.distro().contains("Garden Linux")
+    }
+
+    pub fn kernel_version(&self) -> Box<KernelVersion> {
+        KernelVersion::from_host()
+    }
+
+    pub fn os_id(&self) -> String {
+        self.os_release_key("ID")
+    }
+
+    pub fn build_id(&self) -> String {
+        self.os_release_key("BUILD_ID")
+    }
+
+    pub fn distro(&self) -> String {
+        self.os_release_key("PRETTY_NAME")
+    }
```

Comment:
There are the only 3 fields gathered from `/etc/os-release`, right? Wonder if we could change the HostInfo definition to something like this:
```rust
pub struct HostInfo {
    os: String,
    build: String,
    distro: Stirng,
}
```

Then, instead of reading the file every time the struct is accessed, you could read it once in a constructor method like:
```rust
    pub fn new<T: Into<PathBuf>>(path: T) -> HostInfo {
        let mut os = String::new();
        let mut build = String::new();
        let mut distro = String::new();

        if let Ok(file) = File::open(path.into()) {
            let reader = BufReader::new(file);

            for line in reader.lines() {
                match lines {
                    Ok(s) if s.starts_with("ID") => os = s["ID".len()..].to_owned(),
                    Ok(s) if s.starts_with("BUILD_ID") => build = s["BUILD_ID".len()..].to_owned(),
                    Ok(s) if s.starts_with("PRETTY_NAME") => distro = s["PRETTY_NAME".len()..].to_owned(),
                    Ok(_) => {} // Nothing to do with this line.
                    Err(_) => break,
                }
            }
        }
        HostInfo{os, build, distro}
    }